core/app.go

```diff
@@ -373,3 +376,37 @@ func (a *App) stopCoprocesses() {
 		coprocess.Stop()
 	}
 }
+
+// getStatusHandler generates HTTP response for Telemetry status control plane
+// endpoint.
+func (a *App) getStatusHandler(w http.ResponseWriter, r *http.Request) {
+	// TODO: need to test for GET requests
+	w.WriteHeader(http.StatusOK)
+	w.Header().Set("Content-Type", "application/json")
+
+	servicesJSON, err := json.Marshal(a.Services)
+	if err != nil {
+		log.Errorf("Error generating services for '/status': %v", err)
+	}
+	log.Debugf("Marshaled services: %v", string(servicesJSON))
+
+	// w.Write([]byte(`{ "message": true }`))
+	w.Write(servicesJSON)
+}
+
+// postEnvHandler generates HTTP response for updating environment variables of
+// future CP managed processes.
+func (a *App) postEnvHandler(w http.ResponseWriter, r *http.Request) {
+	// TODO: need to test HTTP method
+	w.WriteHeader(http.StatusOK)
+	w.Header().Set("Content-Type", "application/json")
+	w.Write([]byte(`{ "message": true }`))
+}
+
+func (a *App) handleServer() *control.Server {
+	s := a.Server
+	s.Mux.HandleFunc("/status", a.getStatusHandler)
+	s.Mux.HandleFunc("/v3/env", a.postEnvHandler)
+	s.Serve()
```

Comment:
A little more background on my question. We have two options here and I wasn't sure which you thought was clearer for the code base...

1. We can push the App object/pointer through the call to `s.Serve()` and provide it to the handlers by pushing it through more hoops and function calls...
2. Or... we can simply implement the App handlers on App itself and perform the work close to the existing implementations of `Reload()` and etc.

I'm siding on option 2 but I'm open to suggestions. Also unclear if there's a better Golang primitive or pattern.